flink-runtime/src/main/java/org/apache/flink/streaming/runtime/operators/sink/SinkWriterOperator.java

```diff
@@ -249,13 +276,18 @@ private void emit(
                                 committables.size(),
                                 0)));
         for (CommT committable : committables) {
-            output.collect(
+            emit(
                     new StreamRecord<>(
                             new CommittableWithLineage<>(
                                     committable, checkpointId, indexOfThisSubtask)));
         }
     }
 
+    private void emit(StreamRecord<CommittableMessage<CommT>> message) {
+        output.collect(message);
+        LOG.debug("Sending message to committer: {}", message);
```

Comment:
Nit: Either move the statement before sending or please do a slight rename.
```suggestion
        LOG.debug("Sent message to committer: {}", message);
```